### PR TITLE
State types extensions for tracing

### PR DESCRIPTION
**Pull request by chfast**
This adds some simple inspection methods to `Stack` and `Memory` types needed for tracing.

### CODE REVIEW

File: lib/evmone/execution_state.hpp [hunk 1]
```diff
@@ -83,6 +83,7 @@ class Memory
 
     uint8_t& operator[](size_t index) noexcept { return m_memory[index]; }
 
+    [[nodiscard]] const uint8_t* data() const noexcept { return m_memory.data(); }
```
Comment (axic) [inline lib/evmone/execution_state.hpp:86]: I wonder if in fact you don't actually want something like `bytes_view substr(size_t start, size_t size) const` here?

**Review by axic — APPROVED** [review]

File: lib/evmone/execution_state.hpp (hunk 1 above)
Comment (chfast) [reply lib/evmone/execution_state.hpp:86]: There might be uses of such pattern, but tracing is not one of these.

Comment (axic) [reply lib/evmone/execution_state.hpp:86]: Well your test was doing this, what will be the use case in tracing?

Comment (chfast) [reply lib/evmone/execution_state.hpp:86]: Dump all memory, e.g. `hex({memory.data(), memory.size()})`.

**Pull request merged**